### PR TITLE
clock_control: stm32: F4 domain clock + fixed domain clock

### PR DESCRIPTION
**Pull request by erwango**
This PR contains 2 changes required for https://github.com/zephyrproject-rtos/zephyr/pull/53258

1- Provide DCKCFGR description on F4 series
2- Add ability to describe in dts fixed domain clocks

### CODE REVIEW

**Review by gautierg-st — CHANGES_REQUESTED** [review]

**Review by gautierg-st — DISMISSED** [review]

File: dts/arm/st/f4/stm32f469.dtsi [hunk 1]
```diff
@@ -10,6 +10,8 @@
 	soc {
 		usbotg_fs: usb@50000000 {
 			num-bidir-endpoints = <6>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00000080>,
+				 <&rcc STM32_SRC_PLL_Q CLK48M_SEL(0)>;
```
[the remark this answers is not in the document]
Comment (erwango) [reply dts/arm/st/f4/stm32f469.dtsi]: Remove, thanks

File: drivers/clock_control/clock_stm32_ll_common.c [hunk 2]
```diff
@@ -242,6 +242,11 @@ static inline int stm32_clock_control_configure(const struct device *dev,
 		return err;
 	}
 
+	if (pclken->enr == NO_SEL) {
```
Comment (FRASTM) [inline drivers/clock_control/clock_stm32_ll_common.c:245]: What is a usecase where pclken->enr is NO_SEL is it a "default value"

File: include/zephyr/dt-bindings/clock/stm32f0_clock.h [hunk 3]
```diff
@@ -73,5 +73,7 @@
 #define USART3_SEL(val)		STM32_CLOCK(val, 3, 18, CFGR3_REG)
 /** BDCR devices */
 #define RTC_SEL(val)		STM32_CLOCK(val, 3, 8, BDCR_REG)
+/** Dummy: Add a specificier when no selection is possible */
+#define NO_SEL			0xFF
```
Comment (FRASTM) [inline include/zephyr/dt-bindings/clock/stm32f0_clock.h:77]: could   `NO_SEL`  be defined only _once_ in the #include <zephyr/drivers/clock_control/stm32_clock_control.h>

Comment (erwango) [reply include/zephyr/dt-bindings/clock/stm32f0_clock.h:77]: Not possible. `drivers/clock_control/stm32_clock_control.h` is not a device tree binding (could not be included in *.dtsi)

File: drivers/clock_control/clock_stm32_ll_common.c (hunk 2 above)
Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_common.c:245]: See commit description:
 USB clock on PLL_Q output on F405 devices is not selectable. But yet it needs to be described in device tree in order to be able to call `clock_control_get_rate`.

**Review by gautierg-st — DISMISSED** [review]

File: dts/arm/st/f4/stm32f410.dtsi [hunk 4]
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/dt-bindings/clock/stm32f4_clock.h>
```
Comment (FRASTM) [inline dts/arm/st/f4/stm32f410.dtsi:7]: See that some other stm32f4x devices have a DCKCFGR2 : stm32f410, f412, f413, f423

**Review by FRASTM — CHANGES_REQUESTED** [review]

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Apart from the comments, LGTM

File: dts/arm/st/f4/stm32f410.dtsi [hunk 5]
```diff
@@ -91,6 +96,13 @@
 			#io-channel-cells = <1>;
 		};
 
+		rng: rng@50060800 {
+			compatible = "st,stm32-rng";
+			reg = <0x50060800 0x400>;
+			interrupts = <80 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00000040>;
```
Comment (FRASTM) [inline dts/arm/st/f4/stm32f410.dtsi:103]: Bit 31 RNGEN: RNG clock enable of the RCC AHB1 peripheral clock enable register : RCC_AHB1ENR

**Review by FRASTM — CHANGES_REQUESTED** [review]

File: dts/arm/st/f4/stm32f437.dtsi [hunk 6]
```diff
@@ -4,97 +4,11 @@
...
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <0x40013400 0x400>;
-			interrupts = <84 5>;
-			status = "disabled";
-		};
-
-		/* spi5 is present on all STM32F437XX SoCs except
-		 * STM32F437vX SoCs. Delete node in stm32f437vX.dtsi.
-		 */
```
[the remark this answers is not in the document]
Comment (erwango) [reply dts/arm/st/f4/stm32f437.dtsi:61]: Fixed

File: dts/arm/st/f4/stm32f437.dtsi [hunk 7]
```diff
@@ -4,97 +4,11 @@
...
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <0x40015000 0x400>;
-			interrupts = <85 5>;
-			status = "disabled";
-		};
-
-		/* spi6 is present on all STM32F437XX SoCs except
-		 * STM32F437vX SoCs. Delete node in stm32f437vX.dtsi.
-		 */
```
[the remark this answers is not in the document]
Comment (erwango) [reply dts/arm/st/f4/stm32f437.dtsi:73]: Fixed

**Review by gautierg-st — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**